test/run_test.py

```diff
@@ -424,21 +430,57 @@ def calculate_shards(num_shards: int, tests: List[str], job_times: Dict[str, Tup
     return sharded_jobs
 
 
-def get_shard(which_shard: int, num_shards: int, tests: List[str]) -> List[str]:
+def pull_job_times_from_S3() -> Dict[str, Tuple[float, int]]:
     if HAVE_BOTO3:
-        s3_reports = get_test_time_reports_from_S3()
+        try:
+            s3_reports = get_test_time_reports_from_S3()
+        except (RuntimeError, RuntimeWarning):
+            print('Ran into issues downloading reports from S3.')
+            s3_reports = []
```

Comment:
It is more intuitive to have it outside, since I'd be try catching the whole function in this new implementation.

Reply:
i dont see any addition you had outside of the original try-catch can induce any new failures. could you explain which one can potentially cause more failure? (this can indicate a hidden bug)